pulsar-broker/src/test/java/org/apache/pulsar/client/impl/ClientCnxTest.java

```diff
@@ -173,11 +188,47 @@ public void testCnxReceiveSendError() throws Exception {
         // producer also can send message
         producerOne.send("test");
         producerTwo.send("test");
+        producerThree.send("test");
         producerTwo.close();
         producerOne.close();
+        producerThree.close();
         client.close();
     }
 
+    @Test
+    public void testCnxReceiveSendErrorWithConfigMultiPerConnection() throws Exception {
+        PulsarClient client = PulsarClient.builder().serviceUrl(lookupUrl.toString())
+                .connectionsPerBroker(1000).build();
+
+        // Create a producer with customized name.
+        final String tp = BrokerTestUtil.newUniqueName(NAMESPACE + "/tp");
+        admin.topics().createNonPartitionedTopic(tp);
+        Producer<String> p = client.newProducer(Schema.STRING).producerName("p1").topic(tp).create();
+
+        // Inject a persistence error.
+        org.apache.pulsar.broker.service.Producer serverProducer = pulsar.getBrokerService().getTopic(tp, false)
+                .join().get().getProducers().values().iterator().next();
+        ServerCnx serverCnx = (ServerCnx) serverProducer.getCnx();
+        serverCnx.getCommandSender().sendSendError(serverProducer.getProducerId(), 1/* sequenceId */,
+                ServerError.PersistenceError, "mocked error");
+        // Wait for the client receives the error.
+        Thread.sleep(1000);
```

Comment:
Is it possible to use Awaitability to check the client has received the error?

Reply:
I added Ping-Pong twice to confirm the client has handled the Error we sent

```
// Wait for the client to receive the error.
// If the client confirmed two Pings, it means the client has handled the PersistenceError we sent.
serverCnx.checkConnectionLiveness().join();
serverCnx.checkConnectionLiveness().join();
```